proj/src/main/webapp/script.js

```diff
@@ -37,12 +38,21 @@ if(btnSignUp){
         auth.useDeviceLanguage();
 
         const promise = auth.createUserWithEmailAndPassword(emailVal, passVal).then(function(){
-            auth.currentUser.updateProfile({displayName: fname.value + " " + lname.value}).then(function(){
+            const user = auth.currentUser;
+            user.updateProfile({
+                displayName: fname.value + " " + lname.value
+                }).then(function(){
                 console.log("display name updated successfully");
-                window.location.replace("chat.html");
+                firebase.database().ref("users/" + auth.currentUser.uid).set({
+                    firstName : fname.value,
+                    lastName : lname.value,
+                    tags : tagStr.value.split(',')
+                }).then(function(){
+                    window.location.replace("chat.html");
+                });
             }).catch(function(){
                 console.log("error updating display name");
```

Comment:
Normally I'd suggest removing these console debug logs, but in this case, maybe you want to log something more descriptive while you debug. Like if the catch function can take a parameter with the error, you can log the error given.

In the finished product, you might want some sort of pop up with a user friendly error that says that an error occurred and suggesting they try again or something.

Reply:
Yup, that's what I was planning. Sounds good!